frontend/src/app/main/ui/ds/utilities/swatch.cljs

```diff
@@ -0,0 +1,108 @@
+
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) KALEIDOS INC
+
+(ns app.main.ui.ds.utilities.swatch
+  (:require-macros
+   [app.main.style :as stl])
+  (:require
+   [cuerdas.core :as str]
+   [rumext.v2 :as mf]))
+
+(def ^:private schema:swatch
+  [:map
+   [:background :string]
+   [:ariaLabel {:optional true} :string]
+   [:format {:optional true} [:enum "square" "rounded"]]
+   [:size {:optional true} [:enum "small" "medium"]]
+   [:active {:optional true} :boolean]
+   [:on-activate {:optional true} fn?]])
+
+(def hsla-regex #"^hsla\((\d{1,3}),\s*(\d{1,3})%,\s*(\d{1,3})%,\s*(0|1|0?\.\d+)\)$")
+(def rgba-regex #"^rgba\((\d{1,3}),\s*(\d{1,3}),\s*(\d{1,3}),\s*(0|1|0?\.\d+)\)$")
+
+(defn- is-image? [background]
+  (boolean
+   (str/starts-with? background "http")))
+
+(defn- is-gradient? [background]
+  (boolean
+   (or
+    (str/starts-with? background "linear-gradient")
+    (str/starts-with? background "radial-gradient"))))
+
+(defn- is-color-opacity? [background]
+  (boolean
+   (or (re-matches hsla-regex background)
+       (re-matches rgba-regex background))))
+
+(defn- extract-color-and-opacity [background]
+  (cond
+
+    (re-matches rgba-regex background)
+    (let [[_ r g b a] (re-matches rgba-regex background)]
+      {:color (str "rgb(" r ", " g ", " b ")")
+       :opacity (js/parseFloat a)})
+
+    (re-matches hsla-regex background)
+    (let [[_ h s l a] (re-matches hsla-regex background)]
+      {:color (str "hsl(" h ", " s "%, " l "%)") ;; Strip the alpha
+       :opacity (js/parseFloat a)})
+
+    :else
+    {:color background
+     :opacity 1.0}))
+
+(mf/defc swatch*
+  {::mf/props :obj
+   ::mf/schema schema:swatch}
+  [{:keys [background on-activate format size ariaLabel active] :as props}]
+  (let [format (or format "square")
+        size (or size "small")
+        active (or active false)
+        {:keys [color opacity]} (extract-color-and-opacity background)
+        on-click (when on-activate (mf/use-fn (mf/deps on-activate)
```

Comment:
I think this `on-activate` is weird. This looks like `on-click` and `on-key-down` should actually be delegated properties of a `<button>` (like we're doing with the current buttons in the DS) and set `disabled` if the swatch cannot be interacted with.

Reply:
Fix: this `on-activate` was a poor implementation for the requirement of a dynamic tag in the swatch component.

I've fixed it, setting the appropriate tag for the element depending on the component props. 
In case the author sets an on-click prop on the swatch component, it will become a button, otherwise will remain as presentational with a simple div element.